OperationModeFileWatcherTests can fail with assertion error

The failure doesn't reproduce for me locally.

Link to the build:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob+fast+part2/358/console

Command to reproduce:

```
./gradlew :x-pack:plugin:core:test --tests "org.elasticsearch.license.OperationModeFileWatcherTests.testUpdateModeFromFile" \
  -Dtests.seed=553B2DE4C5A7DD90 \
  -Dtests.security.manager=true \
  -Dtests.locale=ne \
  -Dtests.timezone=UTC \
  -Dcompiler.java=12 \
  -Druntime.java=11

```
Relevant excerpt from the logs:
```
06:37:06   2> REPRODUCE WITH: ./gradlew :x-pack:plugin:core:test --tests "org.elasticsearch.license.OperationModeFileWatcherTests.testUpdateModeFromFile" -Dtests.seed=553B2DE4C5A7DD90 -Dtests.security.manager=true -Dtests.locale=ne -Dtests.timezone=UTC -Dcompiler.java=12 -Druntime.java=11
06:37:06   2> java.lang.AssertionError: 
06:37:06     Expected: <BASIC>
06:37:06          but: was <GOLD>
06:37:06         at __randomizedtesting.SeedInfo.seed([553B2DE4C5A7DD90:B99EDD80697A288D]:0)
06:37:06         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
06:37:06         at org.junit.Assert.assertThat(Assert.java:956)
06:37:06         at org.junit.Assert.assertThat(Assert.java:923)
06:37:06         at org.elasticsearch.license.OperationModeFileWatcherTests.testUpdateModeFromFile(OperationModeFileWatcherTests.java:73)
```

--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/license/OperationModeFileWatcher.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/license/OperationModeFileWatcher.java
@@ -91,6 +91,7 @@ public final class OperationModeFileWatcher implements FileChangesListener {
 
     private synchronized void onChange(Path file) {
         if (file.equals(licenseModePath)) {
+            final OperationMode savedOperationMode = this.currentOperationMode;
             OperationMode newOperationMode = defaultOperationMode;
             try {
                 if (Files.exists(licenseModePath)
@@ -116,11 +117,13 @@ public final class OperationModeFileWatcher implements FileChangesListener {
                     }
                 }
             } finally {
-                // set this after the fact to prevent that we are jumping back and forth first setting to defautl and then reading the
+                // set this after the fact to prevent that we are jumping back and forth first setting to default and then reading the
                 // actual op mode resetting it.
                 this.currentOperationMode = newOperationMode;
             }
-            onChange.run();
+            if (savedOperationMode != newOperationMode) {
+                onChange.run();
+            }
         }
     }
 }

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/license/OperationModeFileWatcherTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/license/OperationModeFileWatcherTests.java
@@ -51,7 +51,7 @@ public class OperationModeFileWatcherTests extends ESTestCase {
     }
 
     public void testInit() throws Exception {
-        onChangeCounter.set(new CountDownLatch(2));
+        onChangeCounter.set(new CountDownLatch(1));
         writeMode("gold");
         assertThat(operationModeFileWatcher.getCurrentOperationMode(), equalTo(License.OperationMode.PLATINUM));
         operationModeFileWatcher.init();

